### ISSUE TITLE
CVE upload fails when "PublicDate: unknown"

### ISSUE THREAD

**Issue by markmorlino** [canonical-web-and-design/ubuntu.com#8045]
*Note - filling in this form will file an issue against the www.ubuntu.com website, not against Ubuntu itself, for Ubuntu bugs, please [go here](https://help.ubuntu.com/community/ReportingBugs)*

## Summary

When uploading CVE files to http[s://ubuntu.com/security/cve, the upload fails when the CVE has a `unknown` as the value for `PublicDate`.

## Process

```
$ ./scripts/publish-cves-to-website-api.py retired/CVE-2006-2192
1 total CVEs
<Response [400]> {"errors":{"0":{"published":["dateutil cannot parse unknown."]}},"message":"Invalid payload"}
```

## Current and expected result

Expectation is that null or something is inserted as the value and the rest of the operation succeeds as with other CVE files:

```
$ ./scripts/publish-cves-to-website-api.py active/CVE-2020-14579
1 total CVEs
<Response [200]> {"created":{},"updated":{"CVE":1,"Package":1,"Status":7}}
```

**Comment by nottrobin** [comment]
Basically, this just means making that field nullable. However, at the moment, I believe the search relies on being able to order CVEs by published date, and I don't know what would happen with this ordering if it was null. We should just make sure that any CVEs with missing published dates are listed at the end (rather than the beginning) when ordering by published date.

**Comment by jpmartinspt** [comment]
The field is nullable on the database, and not required in the API schema. The only issue is that CVE's with null publish dates are currently showing at the top of the table in `/security/cve` but that should be fixed after we merge #8053. 

@markmorlino I tried the following in your branch(tried to MP but have no permission, sorry):
```diff
diff --git a/scripts/publish-cves-to-website-api.py b/scripts/publish-cves-to-website-api.py
index d8d9c174ec..f43f2412e6 100755
--- a/scripts/publish-cves-to-website-api.py
+++ b/scripts/publish-cves-to-website-api.py
@@ -109,7 +109,6 @@ def post_single_cve(cve_filename):
 
     cve = {
         "id": cve_data["Candidate"],
-        "published": cve_data["PublicDate"],
         "description": cve_data["Description"],
         "ubuntu_description": cve_data["Ubuntu-Description"],
         "notes": notes,
@@ -121,6 +120,10 @@ def post_single_cve(cve_filename):
         "status": status,
     }
 
+    published = cve_data.get("PublicDate", "unknown")
+    if published != "unknown":
+        cve["published"] = published
+
     return cve

```
